src/system.jl

```diff
@@ -0,0 +1,302 @@
+"""
+    ServicesTimeSeries(reg, spin, sup_on, sup_off)
+
+Type defining services time series.  Fields are KeyedArray where the keys are generator names
+x datetimes.  Only generators that provide each service are included in the array.
+"""
+struct ServicesTimeSeries
+    reg::KeyedArray{Float64}
+    spin::KeyedArray{Float64}
+    sup_on::KeyedArray{Float64}
+    sup_off::KeyedArray{Float64}
+end
+
+"""
+    Zone(number, reg, spin, sup_on, sup_off)
+
+Type defining a market zone.  The `Zone` is identified by a number.  The other fields contain
+the service requirements for the zone.
+"""
+struct Zone
+    number::Int64
+    reg::Float64
+    spin::Float64
+    sup_on::Float64
+    sup_off::Float64
+end
+
+###### Static Component Types ######
+
+abstract type StaticComponent end
+
+Base.length(components::StaticComponent) = length(getfield(components, 1))
+# define interfaces? Iterator? Table?
```

Comment:
It looks to me like the data has been split up exactly to facilitate the Tables.jl interface, i.e. we split "static" data (table-like) and time-series data, which suggests to me we think there's a use-case for having the "static" data available to work with as a table... Do we have such a use-case? Where e.g. `df = DataFrame(gens::Generators)` would be a useful thing?